plugins/manual_maps/ManualMapLoading.ts

```diff
@@ -0,0 +1,214 @@
+  static async writeMS(curr: tm.CurrentMap, queue: tm.Map[], startAt: number = 0) {
+    const newQueue = (queue.slice(0, config.preloadMaps))
+    if (this.oldQueue !== undefined && this.oldCurr !== undefined
+      && curr.id === this.oldCurr.id && curr.fileName === this.oldCurr.fileName &&
+      this.oldQueue.every(((a, i) => a.id === newQueue[i].id && a.fileName === newQueue[i].fileName))) {
+      Logger.trace("Did not write new MatchSettings")
+      return
+    }
+
+    let maps = newQueue.map(a => `  <challenge>
+    <file>${a.fileName.replaceAll('/', '\\')}</file>
+    <ident>${a.id}</ident>
+  </challenge>
+`)
+    maps.unshift(`  <challenge>
+    <file>${curr.fileName.replaceAll('/', '\\')}</file>
+    <ident>${curr.id}</ident>
+  </challenge>
+`)
+
+    const game = tm.config.game
+    const header: string = `<?xml version="1.0" encoding="utf-8" ?>
+<playlist>
+  <gameinfos>
+    <game_mode>${game.gameMode}</game_mode>
+    <chat_time>10000</chat_time>
+    <finishtimeout>${game.finishTimeout}</finishtimeout>
+    <allwarmupduration>${game.warmUpDuration}</allwarmupduration>
+    <disablerespawn>${game.disableRespawn ? 1 : 0}</disablerespawn>
+    <forceshowallopponents>${game.forceShowOpponents}</forceshowallopponents>
+    <rounds_pointslimit>${game.roundsPointLimitOld}</rounds_pointslimit>
+    <rounds_usenewrules>1</rounds_usenewrules>
+    <rounds_forcedlaps>${game.roundsModeLapsAmount}</rounds_forcedlaps>
+    <rounds_pointslimitnewrules>${game.roundsPointLimitNew}</rounds_pointslimitnewrules>
+    <team_pointslimit>${game.teamPointLimitOld}</team_pointslimit>
+    <team_maxpoints>${game.teamMaxPoints}</team_maxpoints>
+    <team_usenewrules>1</team_usenewrules>
+    <team_pointslimitnewrules>${game.teamPointLimitNew}</team_pointslimitnewrules>
+    <timeattack_limit>${game.timeAttackLimit}</timeattack_limit>
+    <timeattack_synchstartperiod>0</timeattack_synchstartperiod>
+    <laps_nblaps>${game.lapsModeLapsAmount}</laps_nblaps>
+    <laps_timelimit>${game.lapsModeTimeLimit}</laps_timelimit>
+    <cup_pointslimit>${game.cupPointsLimit}</cup_pointslimit>
+    <cup_roundsperchallenge>${game.cupRoundsPerMap}</cup_roundsperchallenge>
+    <cup_nbwinners>${game.cupWinnersAmount}</cup_nbwinners>
+    <cup_warmupduration>${game.cupWarmUpRounds}</cup_warmupduration>
+  </gameinfos>
+  <hotseat>
+    <game_mode>0</game_mode>
+    <time_limit>300000</time_limit>
+    <rounds_count>5</rounds_count>
+  </hotseat>
+  <filter>
+    <is_lan>1</is_lan>
+    <is_internet>1</is_internet>
+    <is_solo>0</is_solo>
+    <is_hotseat>0</is_hotseat>
+    <sort_index>1000</sort_index>
+    <random_map_order>0</random_map_order>
+    <force_default_gamemode>0</force_default_gamemode>
+  </filter>
+  <startindex>${startAt}</startindex>
+`
+    // NOTE: btoa() is deprecated but idk another easy way to do this
+    if (!await Client.call('WriteFile', [{string: 'MatchSettings.trakman.txt'}, {base64: btoa(header + maps.join('') + '</playlist>')}])) {
```

Comment:
Buffer.from(whatever).toString('base64')?

Reply:
Wouldn't that be slower?